src/Azure.Functions.Cli/Actions/KubernetesActions/FunctionMetadataGenerator.cs

```diff
@@ -0,0 +1,174 @@
+                    {"databaseName", properties["database_name"]},
+                    {"containerName", properties["container_name"]},
+                    {"connection", properties["connection"]},
+                    {"leaseContainerName", "leases"},
+                    {"createLeaseContainerIfNotExists", true}
+                });
+                    break;
+                case "event_hub_message_trigger":
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", "eventHubTrigger"},
+                    {"direction", "In"},
+                    {"name", properties.GetValueOrDefault("arg_name", "events")},
+                    {"eventHubName", properties["event_hub_name"]},
+                    {"connection", properties["connection"]},
+                    {"cardinality", "Many"},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
+                    break;
+                case "queue_trigger":
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", "queueTrigger"},
+                    {"direction", "In"},
+                    {"name", properties.GetValueOrDefault("arg_name", "message")},
+                    {"queueName", properties["queue_name"]},
+                    {"connection", properties["connection"]},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
+                    break;
+                case "service_bus_queue_trigger":
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", "serviceBusTrigger"},
+                    {"direction", "In"},
+                    {"name", properties.GetValueOrDefault("arg_name", "message")},
+                    {"queueName", properties["queue_name"]},
+                    {"connection", properties["connection"]},
+                    {"cardinality", "One"},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
+                    break;
+                case "service_bus_topic_trigger":
+                    bindings.Add(new Dictionary<string, object>
+                {
+                    {"type", "serviceBusTrigger"},
+                    {"direction", "In"},
+                    {"name", properties.GetValueOrDefault("arg_name", "message")},
+                    {"topicName", properties["topic_name"]},
+                    {"subscriptionName", properties["subscription_name"]},
+                    {"connection", properties["connection"]},
+                    {"cardinality", "One"},
+                    {"properties", new Dictionary<string, object> { { "supportsDeferredBinding", "True" } } }
+                });
```

Comment:
Checking: should you flatten the nested dictionary with an class (or an entity)?